### DISCUSSION TITLE
TypeError: r is not a function Next.js 14

### DISCUSSION THREAD

**Discussion by jun-fajr** [vercel/next.js#68665] [Help] [answered]
### Summary

I have a streaming website using agora, https://stream-app-one.vercel.app/streaming-detail and I have this problem:

TypeError: r is not a function

![Screenshot 2024-08-08 155841](https://github.com/user-attachments/assets/0b2d97da-1fff-41b6-8a2b-6c3913013bf3)

when i run build in local, there is no error. 
while after deploy there is a problem like that. while build deploy in vercel there is no error

**Comment by KamalSinghKhanna** [comment]
HI, facing the same issue. Did you find any solution?

**Reply by jun-fajr** [reply]
LOL. 
still can't. I've failed for 2 weeks.
can I see your website?
what library and next.js do you use?

**Comment by KamalSinghKhanna** [comment]
I am using agora-react-uikit with nextjs 14.
Actually I have stopped the deployment but agora react sdk was working.

Uikit was giving that error.

**Comment by KamalSinghKhanna** [comment]
@jun-fajr I was able to work around UI kit by downgrading the Next.js to 14.1.4 version.

**Reply by Muhammad-Tayyab2** [reply, marked as answer]
@jun-fajr, I also encountered this issue starting yesterday, and downgrading from Next.js version 14.2.8 to 14.1.4 fixed it for me.

**Comment by MMansy19** [comment]
I have the same error and can't solve it

**Comment by adimaulanat** [comment]
Same here, using agora-rtm-sdk version 1.4.5 and next js 14.2.8

**Reply by adimaulanat** [reply]
Downgrade to 14.1.4 solved the problem

**Comment by vick2592** [comment]
This did not work for me. Did anyone else manage to fix this error somehow. For me it is happening when rendering the squid router widget from my component. This is my repo.

https://github.com/vick2592/CharlieBull

**Comment by Velkamhell97** [comment]
I have the same problem, using  "agora-chat-uikit": "^2.0.0", "next": "^16.0.7". Is there a solution for newer versions of next ?
